src/runtime-rs/crates/shim/src/args.rs

```diff
@@ -52,21 +51,6 @@ impl Args {
         validate::verify_id(&self.id).context("verify container id")?;
         validate::verify_id(&self.namespace).context("verify namespace")?;
 
-        // Ensure `address` is a valid path.
-        let path = PathBuf::from(self.address.clone())
-            .canonicalize()
-            .context(Error::InvalidPath(self.address.clone()))?;
-        let md = path
-            .metadata()
-            .context(Error::FileGetMetadata(format!("{:?}", path)))?;
-        if !md.file_type().is_socket() {
-            return Err(Error::InvalidArgument).context("address is not socket");
-        }
-        self.address = path
-            .to_str()
-            .map(|v| v.to_owned())
-            .ok_or(Error::InvalidArgument)?;
-
```

Comment:
@pmores, just checking if this is possible or not ... but can we warn those instead of erroring out?
By doing this we can make sure at least people will have enough info to debug in case something goes awry in the containerd case.

Reply:
It can definitely be done, I'm just not sure though how useful it would be.  If I get this right, the value is fully owned by the corresponding runtime and its semantics are opaque to a shim.  If we don't know what the value can or cannot be we can't really check it, right?

In particular, cri-o intentionally sends an empty value and is just fine getting it back from the shim.  Warning about the `-address` argument not being a socket would then mean cri-o users being warned of something that's meant to be that way.

How the go shim handles this could be a useful guideline here. I haven't been able to figure that out with any certainty although I didn't see a warning looking into what I _think_ was the pertinent code.  One thing I know is that it doesn't seem to error out.

Reply:
I fully agree with @pmores and @littlejawa. This address is obviously something that should be considered as an opaque blob by the shim. Dropping any code that makes assumption on its content is definitely a good thing.

Reply:
> I fully agree with @pmores and @littlejawa. This address is obviously something that should be considered as an opaque blob by the shim. Dropping any code that makes assumption on its content is definitely a good thing.

... which make me think that the test code in this file should be adapted to your change. I'll submit a patch based on your PR.

Reply:
That's fine by me.